### DISCUSSION TITLE
Set name servers on the client

### DISCUSSION THREAD

**Discussion by robdyck** [strongswan/strongswan#1372] [Q&A]
After not travelling for two plus years, I am revisiting Strongswan. More to the point, I am connecting a road warrior to the home server. It seems after several updates on Fedora, now the name servers cannot be set on the client. Crtically domain names cannot be resolved. I have a suspicion about Fedora's resolver, systemd-resolved. It seems that systemd-resolved can be configured for three modes of operation. Two of the modes involved creating a symlink at /etc/resolv.conf. The link can point at either /run/systemd/resolve/resolv.conf or /run/systemd/stub-resolv.conf. Both I believe are managed by systemd-resolved. The third method is to simply use an old fashioned /etc/resolv.conf. The stub resolver is the system's installed default.

None of this helps as far as the errors are concerned but with the old fashioned /etc/resolv.conf, I can at least resolve domain names. But my preferred name servers are never set.

The error messages implicate resolvconf. On Fedora, invoking resolvconf actually starts resolvectl and supports a subset of resolvconf switches. Resolvectl itself gives a status of systemd-resolved and can be used for queries and set name servers.

Have I missed something? Does anyone have a working configuration for Fedora? Is the Fedora version of resolvconf not compatible with strongswan?

[resolvconf-fails.txt](https://github.com/strongswan/strongswan/files/9864274/resolvconf-fails.txt)

**Comment by tobiasbrunner** [comment]
Is `systemd-resolved` enabled and running? (`resolvconf` as implemented by `resolvectl` will definitely need that. Maybe try `systemctl enable --now systemd-resolved`)

**Reply by robdyck** [reply]
It was started and restarted many times as I experimented.
Good news however. It seems this has come up before and some searches revealed a solution. I installed openresolv. It is available on Fedora. resolvconf is symlinked to /etc/alternatives/resolvconf which is symlinked to /usr/sbin/resolvconf.openresolv. It's a shell script.
Works fine. I restored Fedora to the stub resolver.
